application/server/server.go

```diff
@@ -477,28 +477,42 @@ func textDocumentDidOpenHandler() jrpc2.Handler {
 	})
 }
 
+func isDotSnykFile(uri sglsp.DocumentURI) bool {
+	return strings.HasSuffix(string(uri), ".snyk")
+}
+
+func dotSnykFileDidChangeScan(ctx context.Context, workspaceFolderPath *workspace.Folder) {
+	go workspaceFolderPath.ScanFolder(ctx)
+}
+
 func textDocumentDidSaveHandler() jrpc2.Handler {
 	return handler.New(func(_ context.Context, params sglsp.DidSaveTextDocumentParams) (any, error) {
 		// The context provided by the JSON-RPC server is canceled once a new message is being processed,
 		// so we don't want to propagate it to functions that start background operations
 		bgCtx := context.Background()
-		c := config.CurrentConfig()
-		logger := c.Logger().With().Str("method", "TextDocumentDidSaveHandler").Logger()
+		cfg := config.CurrentConfig()
+		logger := cfg.Logger().With().Str("method", "TextDocumentDidSaveHandler").Logger()
 
 		logger.Info().Interface("params", params).Msg("Receiving")
 		di.FileWatcher().SetFileAsSaved(params.TextDocument.URI)
 		filePath := uri.PathFromUri(params.TextDocument.URI)
 
 		f := workspace.Get().GetFolderContaining(filePath)
-		autoScanEnabled := config.CurrentConfig().IsAutoScanEnabled()
-		if f != nil && autoScanEnabled {
-			go f.ScanFile(bgCtx, filePath)
-		} else {
+
+		if isDotSnykFile(params.TextDocument.URI) {
+			dotSnykFileDidChangeScan(bgCtx, f)
+			return nil, nil
+		}
+
+		autoScanEnabled := cfg.IsAutoScanEnabled()
+		if f != nil {
 			if autoScanEnabled {
-				logger.Warn().Str("documentURI", filePath).Msg("Not scanning, file not part of workspace")
+				go f.ScanFile(bgCtx, filePath)
 			} else {
 				logger.Warn().Msg("Not scanning, auto-scan is disabled")
 			}
+		} else if autoScanEnabled {
```

Comment:
why is this dependent on the autoscan boolean?

Reply:
This bit was just a small refactor to reduce one level of nesting. It's the first [commit](https://github.com/snyk/snyk-ls/pull/563/commits/06891fae540bf5799aed5037a00cc932c2cb37ce)